File: src/main/java/poc/PersonService.java
package poc;

import poc.model.Person;
import poc.repository.PersonRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
public class PersonService {
    private final PersonRepository repository;

    @Autowired
    public PersonService(PersonRepository repository) {
        this.repository = repository;
    }

    @Transactional
    public Person savePerson(Person person) {
        return repository.save(person);
    }

    @Transactional(readOnly = true)
    public List<Person> getAllPeople() {
        return repository.findAll();
    }

    @Transactional(readOnly = true)
    public Optional<Person> getPersonById(int id) {
        return repository.findById(id);
    }
}

File: src/main/kotlin/poc/service/PersonService.kt
package poc.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import poc.model.Person
import poc.repository.PersonRepository
import java.util.*

@Service
class PersonService(@Autowired private val repository: PersonRepository) {
    fun savePerson(person: Person): Person = repository.save(person)

    fun getAllPeople(): List<Person> = repository.findAll()

    fun getPersonById(id: Int): Optional<Person> = repository.findById(id)
}
